veggieseasons/lib/widgets/veggie_headline.dart

```diff
@@ -72,7 +72,9 @@ class VeggieHeadline extends StatelessWidget {
     final themeData = CupertinoTheme.of(context);
 
     return GestureDetector(
-      onTap: () => DetailsScreen.show(Navigator.of(context), veggie.id),
+      onTap: () {
+        context.go('${GoRouter.of(context).location}/details/${veggie.id}');
```

Comment:
Allows to dynamically call this from either `/favorites` or `/search`, and builds `/favorites/details/1` or `/search/details/1`

Reply:
Yeah, go_router doesn't have support for relative routes (https://github.com/flutter/flutter/issues/108177), it might be a good idea to add a comment here to explain what this line is doing.